frontends/web/src/hooks/backbutton.ts

```diff
@@ -0,0 +1,60 @@
+/**
+ * Copyright 2024 Shift Crypto AG
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { useContext, useEffect, useRef } from 'react';
+import { BackButtonContext, THandler } from '@/contexts/BackButtonContext';
+
+/*
+ * The Android back button will call this handler while this hook is active.
+ * The handler can perform an action and:
+ * - return false to stop any further action
+ * - return true for Android to perform the default back operation, which is going back in browser
+ * history if possible, or prompting to quit the app.
+*/
+export const useBackButton = (handler: THandler) => {
+  const { pushHandler, popHandler } = useContext(BackButtonContext);
+
+  // We don't want to re-trigger the handler effect below when the handler changes, no need to
+  // repeat the push/pop pair unnecessarily.
+  const handlerRef = useRef<THandler>(handler);
+  useEffect(() => {
+    handlerRef.current = handler;
+  }, [handler]);
+
+  useEffect(() => {
+    pushHandler(handlerRef.current);
+    return popHandler;
```

Comment:
popHandler currently removes the last handler in the internal array... but I wonder if for any reason there is more than one useBackButton handlers at the same time and some mount and unmount if just removing the last one is always correct.

Alternative which seems more save would be to pass the handler also to popHandler and then let the function search for this exact function and remove just that funciton from the array.

We have something similar here: https://github.com/BitBoxSwiss/bitbox-wallet-app/blob/32bc3d6f4f91af7cb21abb5391b3263687f50e12/frontends/web/src/utils/event.ts#L81-L82